include/experimental/__p1673_bits/blas1_dot.hpp

```diff
@@ -100,10 +101,18 @@ Scalar dot(
                 v1.static_extent(0) == v2.static_extent(0));
 
   using size_type = std::common_type_t<SizeType1, SizeType2>;
-  for (size_type k = 0; k < v1.extent(0); ++k) {
-    init += v1(k) * v2(k);
-  }
-  return init;
+  using scalar_type = std::common_type_t<ElementType1, ElementType2, Scalar>;
```

Comment:
This is not necessary the right type.  For example, if `operator*` returns a higher-precision type than its inputs (as might make sense for custom integer or fixed-point real types, for example), then what you want here is the common type of `Scalar` and the result of `operator*`.  However, it turns out that you don't need `scalar_type` here; please see the comment below.